core/src/main/scala/spinal/core/Vec.scala

```diff
@@ -170,15 +172,39 @@ class Vec[T <: Data](val dataType: HardType[T], val vec: Vector[T]) extends Mult
     }
 
 
+    val dimPattern = "(.*)_([0-9]+)dim"r
+    val (vecName, dimId) = this.getName() match {
+      case dimPattern(vecName, dimId) => (vecName, (dimId.toInt + 1).toString)
+      case x:String => (x, "0")
+      case _ => 
+    }
+
     val ret = dataType()
+    ret match {
+      case ret: Vec[_] => {
+        ret.setWeakName(s"${vecName}_${dimId}dim")
+      }
+      case _ => 
+    }
     val retFlatten = ret.flatten
     for(i <- 0 until vecTransposed.length){
       val target = retFlatten(i)
+      target.setCompositeName(this, s"mux${i}", true)
       target match {
         case bv: BitVector => bv.unfixWidth()
         case _             =>
       }
-      target.assignFrom(target.newMultiplexer(finalAddress, vecTransposed(i)))
+      //target.assignFrom(target.newMultiplexer(finalAddress, vecTransposed(i)))
```

Comment:
This change is problematic, i mean, i can add many many many always blocks in the generated code.

I see the issue you were reffering to in the Verilog backend, which force the newMultiplexed to define a intermediate signal. I'm currently thinking about adding some backend logic to automaticaly propagate name from target signal to the anonymus stuff which drive it.